extensions/kubectl/command.go

```diff
@@ -60,11 +64,12 @@ func Apply(client *rancher.Client, cluster *apisV1.Cluster, yamlContent *managem
 
 	jobTemplate := workloads.NewJobTemplate(jobName, Namespace)
 
-	initContainerCommand := []string{"sh", "-c", fmt.Sprintf("echo \"%s\" > /config/my-pod.yaml", strings.ReplaceAll(yamlContent.YAML, "\"", "\\\""))}
-	initContainer := workloads.NewContainer("copy-yaml", imageSetting.Value, corev1.PullAlways, initVolumeMount, nil, initContainerCommand, nil, nil)
-	jobTemplate.Spec.Template.Spec.InitContainers = append(jobTemplate.Spec.Template.Spec.InitContainers, initContainer)
+	if yamlContent != nil {
+		initContainerCommand := []string{"sh", "-c", fmt.Sprintf("echo \"%s\" > /config/my-pod.yaml", strings.ReplaceAll(yamlContent.YAML, "\"", "\\\""))}
+		initContainer := workloads.NewContainer("copy-yaml", imageSetting.Value, corev1.PullAlways, initVolumeMount, nil, initContainerCommand, nil, nil)
+		jobTemplate.Spec.Template.Spec.InitContainers = append(jobTemplate.Spec.Template.Spec.InitContainers, initContainer)
+	}
 
-	command := []string{"kubectl", "apply", "-f", "/root/.kube/my-pod.yaml"}
 	container := workloads.NewContainer(jobName, imageSetting.Value, corev1.PullAlways, volumeMount, nil, command, securityContext, nil)
```

Comment:
Probably best if we add an error check for command also. if we get an empty string for command, the function fails.

Reply:
Added error check for empty command slice. Thanks!